tools/webots.py

```diff
@@ -174,31 +188,92 @@ def exec_build(target, roles, clean, jobs):
     shutil.rmtree(local_toolchain_dir)
 
 
-def exec_run(role):
+def exec_run(role, num_of_robots=1, sim_address="127.0.0.1"):
+    # Check that image has been built
+    exit_code = subprocess.run(
+        ["docker", "image", "inspect", f"{ROBOCUP_IMAGE_NAME}:{ROBOCUP_IMAGE_TAG}"], stderr=DEVNULL, stdout=DEVNULL
+    ).returncode
+    if exit_code != 0:
+        cprint(
+            f"Image '{ROBOCUP_IMAGE_NAME}:{ROBOCUP_IMAGE_TAG}' not found, please run `./b webots build` to build it",
+            "red",
+            attrs=["bold"],
+        )
+        exit(exit_code)
+
     robocup_logs_dir = os.path.join(b.project_dir, "robocup-logs")
 
     # Ensure the logs directory exists for binding into the container
     os.makedirs(robocup_logs_dir, exist_ok=True)
 
-    docker_run_command = [
-        "docker",
-        "container",
-        "run",
-        "--rm",
-        "--network=host",
-        "--mount",
-        f"type=bind,source={robocup_logs_dir},target=/robocup-logs",
-        "-e",
-        "ROBOCUP_ROBOT_ID=1",
-        "-e",
-        "ROBOCUP_TEAM_COLOR=red",
-        "-e",
-        "ROBOCUP_SIMULATOR_ADDR=127.0.0.1:10001",
-        f"{ROBOCUP_IMAGE_NAME}:{ROBOCUP_IMAGE_TAG}",
-        role,
-    ]
+    process_manager = Manager()
+
+    # Override honcho.manager.terminate() method, this is called by its signal handler on CTRL+C
+    process_manager._killall = exec_stop
+
+    # Add all robot run commands to process_manager
+    for i in range(1, int(num_of_robots) + 1):
+
+        robot_color, port_num = (
+            ("red", 10000 + i) if i <= int(num_of_robots) / 2 else ("blue", 10020 + i - int(int(num_of_robots) / 2))
```

Comment:
However if you set `type=int` in `game_subcommand.add_argument("-r", "--robots", action="store", dest="num_of_robots", required=True)` then you should be confident in `num_of_robots` being an `int` and so the cast should be unnecessary.
https://docs.python.org/3/library/argparse.html#type
```suggestion
            ("red", 10000 + i) if i <= int(num_of_robots // 2) else ("blue", 10020 + i - int(num_of_robots // 2)
```

Reply:
Nice find!